### PR TITLE
Fix some CI failures

### PR DESCRIPTION
**Pull request by giordano**
Or try to.

### CODE REVIEW

File: .github/workflows/UnitTests.yml [hunk 1]
```diff
@@ -467,7 +467,7 @@ jobs:
           # - "1.7"
           # - "nightly"
         MPIWrapper:
-          - "2.8.1"
+          - "2.9.0"
```
Comment (giordano) [inline .github/workflows/UnitTests.yml:470]: This doesn't seem to help: https://github.com/JuliaParallel/MPI.jl/actions/runs/5106808453/jobs/9179210983?pr=734#step:9:326
```
Failed to precompile MPI [da04e1cc-30fd-572f-bb4f-1f8673147195] to "/home/runner/.julia/compiled/v1.9/MPI/jl_qRtHCe".
MPItrampoline: MPI ABI version mismatch:
This version of MPItrampoline requires MPI ABI version 2.9.0, but the loaded MPIwrapper only provides MPI ABI version 2.8.0.
This is MPItrampoline version 5.3.0.
You loaded MPIwrapper version 2.9.0 from file "/usr/local/lib/libmpiwrapper.so".
```
@eschnett  Any suggestion of what we should concretely do here?

Comment (eschnett) [reply .github/workflows/UnitTests.yml:470]: Can you try MPIWrapper 2.10.x? The current version is 2.10.3.

Comment (giordano) [reply .github/workflows/UnitTests.yml:470]:
```suggestion
          - "2.10.3"
```

File: .github/workflows/UnitTests.yml [hunk 2]
```diff
@@ -536,7 +536,7 @@ jobs:
...
```
Comment (giordano) [inline .github/workflows/UnitTests.yml:539]:
```suggestion
          - "2.10.3"
```

**Comment by giordano** [comment]
Ok, this is now less bad than before. MVAPICH job is still failing, but this time inside the tests, at least compilation works.

**Pull request merged**